clients/go/gengo.go

```diff
@@ -381,10 +381,15 @@ func (c *WagClient) do%sRequest(ctx context.Context, req *http.Request, headers
 	}
 	resp, err := c.requestDoer.Do(client, req)
 	if err != nil {
+		c.logger.ErrorD("client-request-finished", map[string]string{
```

Comment:
do we want to put error in the title so it's clear that it's this is logged on errors.

Also, when we get a 50x, err will be nil, but resp.StatusCode will be >= 500. Do we want to track those too?